### PR TITLE
Orchestrator service in substrate template (POC)

### PR DESCRIPTION
**Pull request by nanocryk**
Adds a new subcommand in Substrate template node to connect to another RPC endpoint as an `OrchestratorChainInterface` and fetch some data as proof of concept.

Launch arguments exemple:
```
./target/release/container-chain-template-simple-node rpc-provider --assignement-account 5FscB4qRhFfH61zH5TR5zfbw7X8jyCqT1ccjQgmkwitYyWUf --orchestrator-endpoints wss://fraa-dancebox-rpc.a.dancebox.tanssi.network
```

### CODE REVIEW

File: container-chains/templates/simple/node/src/cli.rs [hunk 1]
```diff
@@ -181,3 +186,16 @@ impl CliConfiguration for BuildSpecCmd {
...
+#[group(skip)]
+pub struct RpcProviderSubcommand {
+    /// Endpoints to connect to orchestrator nodes, avoiding to start a local orchestrator node.
+    /// If this list is empty, a local embeded orchestrator node is started.
+    #[arg(long)]
+    pub orchestrator_endpoints: Vec<Url>,
+
+    /// Account associated with the node, whose assignements will be followed to provide RPC services.
+    #[arg(long)]
+    pub assignement_account: dp_core::AccountId,
```
Comment (girazoki) [inline container-chains/templates/simple/node/src/cli.rs:200]: we probably dont need this account as we can take it from the node key, but let's leave it for now yes

File: container-chains/templates/simple/node/src/command.rs [hunk 2]
```diff
@@ -282,6 +284,43 @@ pub fn run() -> Result<()> {
...
+                };
+
+                // POC: Try to fetch some data through the interface.
+                task_manager
+                    .spawn_handle()
+                    .spawn("rpc_provider_exemple", None, async move {
+                        let mut stream = client.new_best_notification_stream().await.unwrap();
+
+                        while let Some(header) = stream.next().await {
+                            println!("New best block: {}", header.hash());
```
Comment (girazoki) [inline container-chains/templates/simple/node/src/command.rs:317]:
```suggestion
                            log::info!("New best block: {}", header.hash());
```

Comment (girazoki) [reply container-chains/templates/simple/node/src/command.rs:317]: this is where, for every new header, we would ask whether we have being assigned to a new chain right?

**Review by girazoki — APPROVED** [review]
good for me @nanocryk, good job!

**Review by tmpolaczyk — APPROVED** [review]

**Pull request merged**